src/Hooks.tsx

```diff
@@ -50,4 +50,95 @@ function useBrowserStorage<T>(
   return [storedValue, setValue]
 }
 
-export default useBrowserStorage
+// useCookieStorage mirrors useBrowserStorage, but for cookies
+export function useCookieStorage<T>(
+  key: string,
+  initialValue: T,
+  prefix?: string,
+  baseDomain?: string
+): [T, (v: T) => void] {
+  key = `${prefix ?? ''}${key}`
+
+  const getBaseDomain = (): string => {
+    if (baseDomain) return baseDomain
+
+    const domainParts = window.location.hostname.split('.')
+    return domainParts.length > 1
+      ? `${domainParts[domainParts.length - 2]}.${domainParts[domainParts.length - 1]}`
+      : domainParts[0]
+  }
+
+  const getCookie = (name: string): string | null => {
+    const value = `; ${document.cookie}`
+    const parts = value.split(`; ${name}=`)
+    if (parts.length === 2) return parts.pop()?.split(';').shift() ?? null
+    return null
+  }
+
+  const deleteCookie = (name: string) => {
+    document.cookie = `${name}=; path=/; domain=.${getBaseDomain()}; max-age=0`
+  }
+
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    if (initialValue === undefined) {
+      return initialValue
+    }
+
+    const item = getCookie(key)
+    try {
+      return item ? JSON.parse(item) : initialValue
+    } catch (error: any) {
+      console.warn(
+        `Failed to parse stored value for '${key}'.\nContinuing with default value.\nError: ${error.message}`
+      )
+      return initialValue
+    }
+  })
+
+  const setValue = (value: T | ((val: T) => T)): void => {
+    if (value === undefined) {
+      setStoredValue(value as any)
+      deleteCookie(key)
+      return
+    }
+    try {
+      const valueToStore = value instanceof Function ? value(storedValue) : value
```

Comment:
What exactly is happening here? When would you want to store a function in a cookie? And does it call the function (`value`) on the previous value (`storedValue`)?